drivers/power/ltc3350/ltc3350.h

```diff
@@ -0,0 +1,244 @@
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef __LTC3350_H__
+#define __LTC3350_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <string.h>
+#include "no_os_util.h"
+#include "no_os_i2c.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/* LTC3350   Version identification*/
+#define LTC3350_VERSION                  		"1.1.1"
+#define LTC3350_MAJOR                    		1
+#define LTC3350_MINOR                    		1
+#define LTC3350_PATCH                    		2
+
+/* LTC3350 Register Map */
+#define LTC3350_AD_CLR_ALARMS                   0x00
+#define LTC3350_AD_MSK_ALARMS                   0x01
+#define LTC3350_AD_MSK_MON_STATUS               0x02
+#define LTC3350_AD_CAP_ESR_PER                  0x04
+#define LTC3350_AD_VCAPFB_DAC                   0x05
+#define LTC3350_AD_VSHUNT                       0x06
+#define LTC3350_AD_CAP_UV_LVL                   0x07
+#define LTC3350_AD_CAP_OV_LVL                   0x08
+#define LTC3350_AD_GPI_UV_LVL                   0x09
+#define LTC3350_AD_GPI_OV_LVL                   0x0A
+#define LTC3350_AD_VIN_UV_LVL                   0x0B
+#define LTC3350_AD_VIN_OV_LVL                   0x0C
+#define LTC3350_AD_VCAP_UV_LVL                  0x0D
+#define LTC3350_AD_VCAP_OV_LVL                  0x0E
+#define LTC3350_AD_VOUT_UV_LVL                  0x0F
+#define LTC3350_AD_VOUT_OV_LVL                  0x10
+#define LTC3350_AD_IIN_OC_LVL                   0x11
+#define LTC3350_AD_ICHG_UC_LVL                  0x12
+#define LTC3350_AD_DTEMP_COLD_LVL               0x13
+#define LTC3350_AD_DTEMP_HOT_LVL                0x14
+#define LTC3350_AD_ESR_HI_LVL                   0x15
+#define LTC3350_AD_CAP_LO_LVL                   0x16
+#define LTC3350_AD_CTL_REG                      0x17
+#define LTC3350_AD_NUM_CAPS                     0x1A
+#define LTC3350_AD_CHRG_STATUS                  0x1B
+#define LTC3350_AD_MON_STATUS                   0x1C
+#define LTC3350_AD_ALARM_REG                    0x1D
+#define LTC3350_AD_MEAS_CAP                     0x1E
+#define LTC3350_AD_MEAS_ESR                     0x1F
+#define LTC3350_AD_MEAS_VCAP1                   0x20
```

Comment:
You could modify these 4 reg addresses so that the implementation simplifies:

#define LTC3350_AD_MEAS_VCAP_X(x)		        ((0x20) + (x) - (1))